File: rgctl-tests/ecommerce-csharp/src/Ecommerce/Controllers/CartController.cs
using Ecommerce.Dto;
using Ecommerce.Services;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

namespace Ecommerce.Controllers;

[ApiController]
[Authorize]
[Route("api/cart")]
public class CartController : ControllerBase
{
    private readonly CartService _cartService;

    public CartController(CartService cartService)
    {
        _cartService = cartService;
    }

    [HttpGet]
    public async Task<ActionResult<CartDto>> GetCart()
    {
        return Ok(await _cartService.GetCartAsync());
    }

    [HttpPost("items")]
    public async Task<ActionResult<CartDto>> AddItem(
        [FromQuery] long productId,
        [FromQuery] int quantity = 1)
    {
        return Ok(await _cartService.AddItemAsync(productId, quantity));
    }

    [HttpDelete("items/{productId:long}")]
    public async Task<ActionResult<CartDto>> RemoveItem(long productId)
    {
        return Ok(await _cartService.RemoveItemAsync(productId));
    }
}

File: rgctl-tests/ecommerce-java/src/main/java/com/example/ecommerce/controller/CartController.java
package com.example.ecommerce.controller;

import com.example.ecommerce.dto.CartDto;
import com.example.ecommerce.service.CartService;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/cart")
public class CartController {

    private final CartService cartService;

    public CartController(CartService cartService) {
        this.cartService = cartService;
    }

    @GetMapping
    public CartDto getCart() {
        return cartService.getCart();
    }

    @PostMapping("/items")
    public CartDto addItem(@RequestParam Long productId, @RequestParam(defaultValue = "1") int quantity) {
        return cartService.addItem(productId, quantity);
    }

    @PutMapping("/items/{productId}")
    public CartDto updateItem(@PathVariable Long productId, @RequestParam int quantity) {
        return cartService.updateItem(productId, quantity);
    }

    @DeleteMapping
    public CartDto clearCart() {
        return cartService.clearCart();
    }
}
